Sources/Testing/Traits/ParallelizationTrait.swift

```diff
@@ -16,31 +16,28 @@
 /// test suite, this trait causes that suite to run its contained test functions
 /// and sub-suites serially instead of in parallel.
 ///
-/// This trait is recursively applied: if it is applied to a suite, any
-/// parameterized tests or test suites contained in that suite are also
-/// serialized (as are any tests contained in those suites, and so on.)
+/// If this trait is applied to a suite, any test functions or test suites
+/// contained in that suite are also serialized (as are any tests contained in
+/// those suites, and so on.)
 ///
 /// This trait does not affect the execution of a test relative to its peers or
 /// to unrelated tests. This trait has no effect if test parallelization is
 /// globally disabled (by, for example, passing `--no-parallel` to the
 /// `swift test` command.)
 ///
 /// To add this trait to a test, use ``Trait/serialized``.
-public struct ParallelizationTrait: TestTrait, SuiteTrait {
-  public var isRecursive: Bool {
-    true
-  }
-}
+public struct ParallelizationTrait: TestTrait, SuiteTrait {}
 
-// MARK: - SPIAwareTrait
+// MARK: - TestScoping
 
-@_spi(ForToolsIntegrationOnly)
-extension ParallelizationTrait: SPIAwareTrait {
-  public func prepare(for test: Test, action: inout Runner.Plan.Action) async throws {
-    if case var .run(options) = action {
-      options.isParallelizationEnabled = false
-      action = .run(options: options)
+extension ParallelizationTrait: TestScoping {
+  public func provideScope(for test: Test, testCase: Test.Case?, performing function: @Sendable () async throws -> Void) async throws {
+    guard var configuration = Configuration.current else {
+      throw SystemError(description: "There is no current Configuration when attempting to provide scope for test '\(test.name)'")
```

Comment:
`Configuration` isn't API, so this may be a confusing message. Why not just do `var configuration = Configuration.current ?? .init()` instead?

Reply:
This really would indicate a bug in our implementation of the testing library, so here at least, I wanted to throw an error instead of attempting to recover. The same argument could be made for the usages in `Runner.swift`, but making those `throws` felt more awkward so I did opt to use `?? .init()`.

Reply:
If it would indicate a bug in our code, let's be sure to include the "file a bug" boilerplate in the message.